### DISCUSSION TITLE
focus issues when using useFieldArray + virtualised list

### DISCUSSION THREAD

**Discussion by rameshx** [react-hook-form/react-hook-form#5891] [Q&A] [answered]
I'm using useFieldArray with react-window to build a virtualized list of inputs but I'm facing focus issues.

After submit, the input with validation error is not getting focus.
the inputs are also losing focus when their values are removed or changed (takes one character and loses focus).

Can anyone help me fix this?

code sandbox link: https://codesandbox.io/s/react-hook-form-usefieldarray-virtual-inputs-forked-mdpul?file=/src/App.js

Steps to reproduce:
just clear the value (so that validation fails) of any of the inputs and click submit

**Comment by bluebill1049** [comment, marked as answer]
can you try to use the latest v7?

**Reply by rameshx** [reply]
@bluebill1049 thanks for responding. Is there a workaround in v6? there are several components in our app so, switching to v7 will require me to update all of them

**Reply by bluebill1049** [reply]
Can you try to use Controller with render prop with focus callback? So U can make sure ref is assigned properly.

**Reply by rameshx** [reply]
Thanks, I found a solution. Avoiding inline function definition and using a component for the rows solved it.
https://codesandbox.io/s/react-hook-form-usefieldarray-virtual-inputs-forked-yc4bt?file=/src/App.js

this is working without the focus callback too
https://codesandbox.io/s/react-hook-form-usefieldarray-virtual-inputs-forked-5wy8x?file=/src/App.js

**Reply by bluebill1049** [reply]
awesome stuff.
